aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -17,22 +17,32 @@ use move_binary_format::{errors::*, CompiledModule};
 use move_core_types::{
     account_address::AccountAddress,
     language_storage::{ModuleId, StructTag},
+    metadata::Metadata,
     resolver::{ModuleResolver, ResourceResolver},
     vm_status::StatusCode,
 };
 use move_table_extension::{TableHandle, TableResolver};
-use move_vm_runtime::move_vm::MoveVM;
-use std::{
-    collections::BTreeMap,
-    ops::{Deref, DerefMut},
-};
+use std::{collections::BTreeMap, ops::Deref};
 
-pub(crate) fn get_any_resource(
+pub(crate) fn get_resource_group_from_metadata(
+    struct_tag: &StructTag,
+    metadata: &[Metadata],
+) -> Option<StructTag> {
+    let metadata = aptos_framework::get_metadata(metadata);
+    metadata?
```

Comment:
nit: I'll put the ? inline with the above line

Reply:
nice spot

Reply:
I must have accidentilly reverted it with a cmd-z. Fixed